### DISCUSSION TITLE
MQTT default disabled?

### DISCUSSION THREAD

**Discussion by jonofmac** [HASwitchPlate/openHASP#529] [General]
I'm building 0.7.0-rc7 from source because it's the only version that works with my board (makerfabs s3 tft 35 SPI).

I'm having an issue where no matter what I seem to enable, mqtt is reported as being disabled. Is this a bug or something in my config that needs to be turned on? Searchign through the obvious confgiuration files, I see no mention of mqtt outside of predefining the client settings.

![image](https://github.com/HASwitchPlate/openHASP/assets/11986470/8ede09ea-7009-4f55-b0a9-7ac50330da09)

Edit:
I found inside of _esp32.ini that under lib_deps, knolleary/PubSubClient@^2.8.0 was commented out. I uncommented it and then recompiled it. So it does indead appear that mqtt client is default disabled, but uncomented this did not enable the client for me...

I see pubsubclient in the build dependencies
```
Scanning dependencies...
Dependency Graph
|-- ArduinoJson @ 6.21.3
|-- lvgl @ 7.11.1+sha.156561b
|-- lv_lib_png @ 0.0.0+20230823195238.sha.d42282b
|-- lv_lib_bmp @ 0.0.0+20230823195239.sha.b6ca85d
|-- lv_lib_gif @ 0.0.0+20230823195241.sha.afdc2d6
|-- ConsoleInput @ 0.0.3+sha.79180a9
|-- AceButton @ 1.10.1
|-- StreamUtils @ 1.7.3
|-- PubSubClient @ 2.8.0
|-- LovyanGFX @ 1.1.8
|-- WiFi @ 2.0.0
|-- ArduinoLog
|-- Wire @ 2.0.0
|-- FT6336U CTP Controller @ 1.0.2
|-- SPI @ 2.0.0
|-- Arduino_RPi_DPI_RGBPanel_mod
|-- Preferences @ 2.0.0
|-- EEPROM @ 2.0.0
|-- lv_fs_if @ 0.0.0+20220202035348
|-- HTTPClient @ 2.0.0
|-- CharStream
|-- StringStream
|-- freetype @ 2.10.4
|-- lv_lib_freetype
|-- FS @ 2.0.0
|-- lv_lib_qrcode
|-- WiFiClientSecure @ 2.0.0
|-- SimpleFTPServer @ 2.1.5
|-- Update @ 2.0.0
|-- WebServer @ 2.0.0
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- ArduinoOTA @ 2.0.0
|-- HTTPUpdate @ 2.0.0
|-- Ethernet @ 2.0.0
|-- ETHSPI
|-- LittleFS @ 2.0.0
|-- SPIFFS @ 2.0.0
Building in release mode
```

**Comment by fvanroie** [comment]
What does your configuration of the [MQTT Settings](https://www.openhasp.com/0.7.0/firmware/configuration/mqtt/) look like? The MQTT Client only gets configured if the MQTT server and port are set.

The Pubsub client library is **not** needed anymore, since 0.7.0 we use the esp_mqtt client instead.

**Comment by jonofmac** [comment]
I modified the user_config_override.h file to include my full mqtt broker settings.

![image](https://github.com/HASwitchPlate/openHASP/assets/11986470/01904857-3fba-4ed4-a0ab-0f971fb280ee)

Then I uncommented the use config override note in platformio_override.ini
![image](https://github.com/HASwitchPlate/openHASP/assets/11986470/4a6e52d8-5537-46a9-ab97-4475ec946310)


However, on a fresh boot, my mqtt settings in the web app are not fully populated:
![image](https://github.com/HASwitchPlate/openHASP/assets/11986470/2032be26-fc7c-468c-8186-e03ab1690e46)



If I manually retype all the settings, and press save, mqtt is still shown as disabled.

**Reply by fvanroie** [reply]
- The settings in the firmware are only used as defaults when there is no filesystem detected. Erase the whole flash and they should be applied at next boot.
- If you save the settings, you still need to soft reboot the plate using the 'Reboot Device' button. Hard powering off/on the device will not work.

**Reply by jonofmac** [reply]
That was it! Thank you. First board I got specifically to try to learn openhasp so I forgot that detail :)

mqtt is now reporting that it's connected and appears to be publishing messages now.
